quinn-proto/src/connection.rs

```diff
@@ -154,6 +154,8 @@ where
     rem_cids: Vec<IssuedCid>,
     /// State of the unreliable datagram extension
     datagrams: DatagramState,
+    /// The number of open *outbound* streams.
+    outbound_streams: u64,
```

Comment:
We should use the standard term "send streams" throughout. I also agree that this probably makes more sense as a member of `Streams`.

Reply:
Agreed.  The difficult part is knowing when to decrement the number of send streams.  I suspect that there are missing calls to `maybe_cleanup`.